chat.go

```diff
@@ -200,7 +200,15 @@ type ChatCompletionRequest struct {
 	// incorrect: `"logit_bias":{"You": 6}`, correct: `"logit_bias":{"1639": 6}`
 	// refs: https://platform.openai.com/docs/api-reference/chat/create#chat/create-logit_bias
 	LogitBias map[string]int `json:"logit_bias,omitempty"`
-	User      string         `json:"user,omitempty"`
+	// LogProbs indicates whether to return log probabilities of the output tokens or not.
+	// If true, returns the log probabilities of each output token returned in the content of message.
+	// This option is currently not available on the gpt-4-vision-preview model.
+	LogProbs *bool `json:"logprobs,omitempty"`
```

Comment:
If we use `*bool` and `*int` here, we'll be unable to set those inline like
```go
req := &openai.ChatCompletionRequest{
	LogProbs: true,
	TopLogProbs: 3,
}
```

See example: https://go.dev/play/p/YZzhc0nEOzf

Using just `bool` and `int` would be more ergonomic for developers and we'll still be able to `omitempty` as `top_logprobs=0` is the same as `null`.

Reply:
Ah yes good shout ✅